darwin/importer/formats/nifti.py

```diff
@@ -0,0 +1,291 @@
+        for i in range(shape[-1]):
+            slice_mask = img[:, :, i].astype(np.uint8)
+            for class_name, class_idxs in processed_class_map.items():
+                frame_annotations = {}
+                if class_name == "background":
+                    continue
+                class_mask = np.isin(slice_mask, class_idxs).astype(np.uint8).copy()
+                polygon = mask_to_polygon(mask=class_mask, class_name=class_name)
+                if polygon is None:
+                    continue
+                frame_annotations[i] = polygon
+                video_annotation = dt.make_video_annotation(
+                    frame_annotations,
+                    keyframes={i: True, i + 1: True},
+                    segments=[[i, i + 1]],
+                    interpolated=False,
+                    slot_names=[],
+                )
+                video_annotations.append(video_annotation)
+    elif mode == "video":  # For each class produce a single video annotation
+        for class_name, class_idxs in processed_class_map.items():
+            if class_name == "background":
+                continue
+            video_annotation = get_video_annotation(img, class_idxs=class_idxs, class_name=class_name)
+            if video_annotation is None:
+                continue
+            video_annotations.append(video_annotation)
+    annotation_classes = set(
+        [dt.AnnotationClass(class_name, "polygon", "polygon") for class_name in class_map.values()]
+    )
+    return dt.AnnotationFile(
+        path=json_path,
+        filename=str(filename),
+        annotation_classes=annotation_classes,
+        annotations=video_annotations,
+    )
+
+
+def get_video_annotation(volume: np.ndarray, class_name: str, class_idxs: List[int]) -> Optional[dt.VideoAnnotation]:
+    frame_annotations = OrderedDict()
+    for i in range(volume.shape[-1]):
+        slice_mask = volume[:, :, i].astype(np.uint8)
+        class_mask = np.isin(slice_mask, class_idxs).astype(np.uint8).copy()
+        if class_mask.sum() == 0:
+            continue
+
+        polygon = mask_to_polygon(mask=class_mask, class_name=class_name)
+        if polygon is None:
+            continue
+        frame_annotations[i] = polygon
+    all_frame_ids = list(frame_annotations.keys())
+    if not all_frame_ids:
+        return None
+    if len(all_frame_ids) == 1:
+        segments = [[all_frame_ids[0], all_frame_ids[0] + 1]]
+    elif len(all_frame_ids) > 1:
+        segments = [[min(all_frame_ids), max(all_frame_ids)]]
+    video_annotation = dt.make_video_annotation(
+        frame_annotations,
+        keyframes={f_id: True for f_id in all_frame_ids},
+        segments=segments,
+        interpolated=False,
+        slot_names=[],
+    )
+    return video_annotation
+
+
+def mask_to_polygon(mask: np.ndarray, class_name: str) -> Optional[dt.Annotation]:
+    _labels, external_paths, _internal_paths = find_contours(mask)
```

Comment:
Is this the same code we have in centermask and clicker?

Reply:
Similar, maybe not exactly the same. Why?

Reply:
It might be wise to turn this into a more generic helper that we can use both here and in the model integrations, but it's ok to keep this out of scope